### ISSUE TITLE
File Ordering

### ISSUE THREAD

**Issue by weiweishi** [ualbertalib/jupiter#362]
This is to track the file ordering issue that is currently blocked by the version of rails and active_storage we are on. The hope is that when we move to the new version (5.2), we will be able to refactor our current approach and display the files in the same order as when they are uploaded, and that is sufficient for the IR phase of the implementation.

**Comment by murny** [comment]
Might be some confusion here. Currently the order the files are uploaded are the exact same way they come out for ActiveStorage/Deposit Item/Draft Items. (However this means if I drag and drop 10 files from my desktop into the upload files form, it depends on which files finish uploading first not the order they were uploaded with). So if this is the issue then there is nothing to fix here. 

With that said, there is currently no way of re-ordering files after upload (aside from...removing the files and re-uploading them in the desired order). If this is the problem, this will still not be fixed with upgrading ActiveStorage that comes with Rails 5.2 as ActiveStorage has no concept of ordering files aside from who got uploaded first (based on their primary keys in the database). So this requires work on our part to figure this out if this is needed.

Now ActiveStorage is just one piece of the puzzle. The bigger concern is Fedora. We can upload and preserve the same order (or heck you can choose whatever order you want, maybe you want to sort them in alphabetical order etc) that user's uploaded the files in ActiveStorage into Fedora (with the addition of adding the chosen thumbnail as the first file? or whatever way we will want to handle the thumbnail for this). But is there a guaranteed way of preserving this order in Fedora? I assume this is where Proxy Objects come in and help with this. But needs investigation.

**Comment by mbarnett** [comment]
> But is there a guaranteed way of preserving this order in Fedora? I assume this is where Proxy Objects come in and help with this. But needs investigation.

Yes. Via Hydra-works and related layers, ordering proxies are created that maintain the file order that the files are added to Fedora in (and you can reorder with it updating the proxies).

**Comment by weiweishi** [comment]
Confirmed with Leah this is not required for launch, but need to be developed in the near future. Put it in post-launch milestone.

**Comment by weiweishi** [comment]
will verify with @leahvanderjagt if we can close this for Jupiter I and revisit file ordering in the refactoring work and jupiter phase 2
